### DISCUSSION TITLE
A way to unlock the custom dlls loaded in a previous c# script execution

### DISCUSSION THREAD

**Discussion by bernatagulloesbrina** [TabularEditor/TabularEditor3#429] [🙋♀️TE3 Q&A]
Hi, 

I'm trying to keep developing my custom dll and there's something that slows the development process

I build the dll,  and I copy the path of the created file to tabular editor and do a test 

if I need to change my dll, however, I can't do it as Tabular Editor is locking the file 

I guess I could copy the file to a different folder each time and put the link to each new folder, but it would be creat if I could use some command telling Tabular Editor to unlock a dll file so I can edit it. 

It would be great if at the end of any execution it would unlock the files used, but if I need to run some command  it's still better than closing and opening tabular editor

**Comment by bernatagulloesbrina** [comment]
Ok this is bizarre 
I tried the approach of copying the dll in a separate folder to be able to at least build again, but somehow Tabular Editor 3 blocks that file --- is that even possible? 

I've closed and open Tabular Editor 3 (no process should be kept running right?) before running any script I was able to build the dll again from Visual Studio 

I copied the dll into a folder on my desktop folder 
and I run this:

`#r "C:\Users\ESAgulloBe\Desktop\dll\1\ExtendedTOMWrapper.dll"

using ExtendedTOMWrapper; 

string myName;
StringUtils.InitString(stringToInit: out myName, checkType: StringUtils.CheckType.Table,model:Model);
            ScriptHelper.Output(myName);`



after that I tried to build again the dll from visual studio , and I get this error error in the output window

> 1>C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Current\Bin\Microsoft.Common.CurrentVersion.targets(4502,5): error MSB3027: Could not copy "obj\x64\Debug\ExtendedTOMWrapper.dll" to "bin\x64\Debug\ExtendedTOMWrapper.dll". Exceeded retry count of 10. Failed. T**he file is locked by: "Tabular Editor 3** (13572)"

> C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Current\Bin\Microsoft.Common.CurrentVersion.targets(4502,5): error MSB3021: Unable to copy file "obj\x64\Debug\ExtendedTOMWrapper.dll" to "bin\x64\Debug\ExtendedTOMWrapper.dll". The process cannot access the file 'bin\x64\Debug\ExtendedTOMWrapper.dll' because it is being used by another process.


I don't know how this can happen, so if a workaround can be found other than closing and opening tabular editor 3 I would be very happy 

Regards!

**Reply by otykier** [reply]
We'll look into it.

**Reply by bernatagulloesbrina** [reply]
thank you!
